cli/template/extras/src/trpc/server.ts

```diff
@@ -1,29 +1,64 @@
 import {
   createTRPCProxyClient,
   loggerLink,
-  unstable_httpBatchStreamLink,
+  TRPCClientError,
 } from "@trpc/client";
+import { callProcedure } from "@trpc/server";
+import { observable } from "@trpc/server/observable";
+import { type TRPCErrorResponse } from "@trpc/server/rpc";
 import { cookies } from "next/headers";
+import { cache } from "react";
 
-import { type AppRouter } from "~/server/api/root";
-import { getUrl, transformer } from "./shared";
+import { appRouter } from "~/server/api/root";
+import { createTRPCContext } from "~/server/api/trpc";
+import { transformer } from "./shared";
 
-export const api = createTRPCProxyClient<AppRouter>({
+/**
+ * This wraps the `createTRPCContext` helper and provides the required context for the tRPC API when
+ * handling a tRPC call from a React Server Component.
+ */
+const createContext = cache(() => {
```

Comment:
i dont love that we have functions with the same `createContext` name in `server.ts` and `route.ts`. but not sure`createRSCContext` is any better.

wanted to bring it up but dont let naming bikeshedding stop us from merging :)